### PR TITLE
Converted AssuredService to H5

### PR DESCRIPTION
**Pull request by BenGalewsky**
- Converted the class to use H5 annotations
- Removed the sequence
- Broke from base class to make the id column more meaningful

While working on this, I'm unable to test. So this represents a best guess while waiting til we get the workflow to this point.

Progress on #36

### CODE REVIEW

File: psm-app/db/seed.sql [hunk 1]
```diff
@@ -76,7 +77,7 @@ CREATE TABLE service_assurance_types (
 CREATE TABLE service_assurance_ext_types (
   code CHARACTER VARYING(2) PRIMARY KEY,
   description TEXT UNIQUE,
-  service_assurance_code TEXT
+  service_assurance_code CHARACTER VARYING(2) REFERENCES service_assurance_types(code)
```
Comment (cecilia-donnelly) [inline psm-app/db/seed.sql]: This doesn't seem to be related to AssuredService -- is it?  It would be cleaner to make this change separately.

File: psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java [hunk 2]
```diff
@@ -3,162 +3,103 @@
  */
 package gov.medicaid.entities;
 
+import javax.persistence.*;
```
Comment (cecilia-donnelly) [inline psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java]: Ah, using `*` gets us `Column`, `Table`, etc?  Is there any reason not to?

File: psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java [hunk 3]
```diff
@@ -3,162 +3,103 @@
...
-     */
+    @Column(name = "effective_date")
     private Date effectiveDate;
 
-    /**
-     * The service assurance type.
-     */
+    @ManyToOne
+    @JoinColumn(name = "service_assurance_code")
     private ServiceAssuranceType type;
```
Comment (cecilia-donnelly) [inline psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java]: Hmm, I'm confused about the difference between this and `extendedTypes`, below.  What's going on with these?  Should they have different column names, or do we just need one of them?

File: psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java (hunk 2 above)
Comment (jasonaowen) [inline psm-app/services/src/main/java/gov/medicaid/entities/AssuredService.java]: Wildcard imports are somewhat subjective. I configured IntelliJ not to do them, as I find that being explicit is more important to me than being terse. The [Twitter Java style guide](https://github.com/twitter/commons/blob/master/src/java/com/twitter/common/styleguide.md#no-wildcard-imports) says:

> Wildcard imports make the source of an imported class less clear.

The [Google Java style guide](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports) and the [OpenJDK style guide](http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-import-statements) both also say to avoid wildcard imports.

All that said, we haven't really talked about this so far, and in this case it's the only import statement, and it's importing from a package that isn't likely to change, so probably it's fine.

File: psm-app/db/seed.sql (hunk 1 above)
Comment (BenGalewsky) [inline psm-app/db/seed.sql]: This is a oneToMany reference from ServiceAssuranceExtType back to AssuredService. I do need to update ServiceAssuranceExtType to have the corresponding manyToOne relationship

**Comment by BenGalewsky** [comment]
This entity is only used as part of the Minnesota Chemical Dependency Program. We don't have seed data for this and it is considered a low priority. I'm going to close this PR, but we will keep the branch around for whenever we get the requirement to convert this entity.
